### ISSUE TITLE
Scan doesn't report assertionFailure and fatalError messages and returns Exit status: 65

### ISSUE THREAD

**Issue by juraj-blahunka** [fastlane/fastlane#12955]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

If a test runs code, which includes `assertionFailure` or `fatalError`, the error isn't reported in scan's output.
The issue was already reported in #6396 but re-appeared.

##### Complete output when running fastlane, including the stack trace and command used

This is what I've got after running `fastlane scan` on an example project [ScanAssertionFailure.zip](https://github.com/fastlane/fastlane/files/515674/ScanAssertionFailure.zip)

```

	 Executed 2 tests, with 1 failure (0 unexpected) in 0.000 (0.000) seconds

** TEST FAILED **
[11:53:44]: Exit status: 65
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 1 |
| Number of failures | 0 |
+--------------------+---+


[!] Test execution failed. Exit status: 65
```

### Environment

<details>
### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.5                                     |
| Ruby                        | 2.4.2                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.16.1                          |
| Installation Source         | ~/.rvm/gems/ruby-2.4.2/bin/fastlane         |
| Host                        | Mac OS X 10.13.5 (17F77)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.2/lib                |
| OpenSSL Version             | OpenSSL 1.1.0g  2 Nov 2017                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.4.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.100.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| executable-hooks          | 1.4.1        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.16.1       |
| io-console                | 0.4.6        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.5        |
| xcodeproj                 | 1.5.8        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| json                      | 2.0.4        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| faraday_middleware        | 0.12.2       |
| naturally                 | 2.1.0        |
| simctl                    | 1.6.4        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.21.2       |
| emoji_regex               | 0.1.1        |
| faraday                   | 0.15.1       |
| openssl                   | 2.0.5        |
| excon                     | 0.62.0       |
| CFPropertyList            | 3.0.0        |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| fastimage                 | 2.1.3        |
| unicode-display_width     | 1.3.2        |
</details>

*generated on:* **2018-07-24**
</details>

**Comment by BabyAzerty** [comment]
Same issue here

**Comment by joshdholtz** [comment]
I also came across this myself this past week 😔  I will look into 💪

**Comment by juraj-blahunka** [comment]
Still seeing this on `2.102.0`

**Comment by juraj-blahunka** [comment]
Seems to be fixed on  `2.106.0`:

```
✗ testThatDoesntGetReported, Test crashed
...
Failing tests:
ScanAssertionFailureTests.testThatDoesntGetReported()
** TEST FAILED **
[09:08:23]: Exit status: 65
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 2 |
| Number of failures | 1 |
+--------------------+---+
```
